dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/client/HttpClientResponseTracingHandler.java

```diff
@@ -35,14 +37,22 @@ public void channelRead(final ChannelHandlerContext ctx, final Object msg) {
         if (finishSpan) {
           Tags.ERROR.set(span, Boolean.TRUE);
           span.log(Collections.singletonMap(ERROR_OBJECT, throwable));
-          Tags.HTTP_STATUS.set(span, 500);
+          int status;
+          try {
```

Comment:
Simpler to just to set the tag in the try:

```java
try {
  int status = ((HttpResponse) msg).getStatus().code();
  NettyUtils.setClientSpanHttpStatus(span, status);
} catch (final Throwable ex) {
  // Ignored.
}
```

Is the behavior change (compared to prior which always set 500) expected though?

Reply:
I wanted not to hide exceptions from the status setting utility, but agree it's simpler.  @keitwb and I agreed it's better not to fabricate a status code in the client instrumentation.

Reply:
updated